lib/Runtime/Library/Utf8String.h

```diff
@@ -0,0 +1,142 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+#include "Codex/Utf8Codex.h"
+
+namespace Js
+{
+    class Utf8String : public JavascriptString
+    {
+    private:
+        typedef struct {
+            FieldNoBarrier(size_t) length;
```

Comment:
Why size_t not charcount_t?

Reply:
This is the length of the utf8 string, which can be up to 3 times as long as the same string in utf16. Since a utf16 string can be up to  ~2^31 in chakra, the corresponding utf8 string may be more than 2^32 in length, so utf8 lengths need to be size_t

Reply:
good point, thanks

---
In reply to: [205622942](https://github.com/Microsoft/ChakraCore/pull/5348#discussion_r205622942) [](ancestors = 205622942)